crates/ruff_workspace/src/configuration.rs

```diff
@@ -1132,6 +1135,201 @@ pub fn resolve_src(src: &[String], project_root: &Path) -> Result<Vec<PathBuf>>
+    }
+
+    if top_level_options.flake8_bandit.is_some() {
+        used_options.push("flake8-bandit");
+    }
+
+    if top_level_options.flake8_bugbear.is_some() {
+        used_options.push("flake8-bugbear");
+    }
+
+    if top_level_options.flake8_builtins.is_some() {
+        used_options.push("flake8-builtins");
+    }
+
+    if top_level_options.flake8_comprehensions.is_some() {
+        used_options.push("flake8-comprehensions");
+    }
+
+    if top_level_options.flake8_copyright.is_some() {
+        used_options.push("flake8-copyright");
+    }
+
+    if top_level_options.flake8_errmsg.is_some() {
+        used_options.push("flake8-errmsg");
+    }
+
+    if top_level_options.flake8_quotes.is_some() {
+        used_options.push("flake8-quotes");
+    }
+
+    if top_level_options.flake8_self.is_some() {
+        used_options.push("flake8-self");
+    }
+
+    if top_level_options.flake8_tidy_imports.is_some() {
+        used_options.push("flake8-tidy-imports");
+    }
+
+    if top_level_options.flake8_type_checking.is_some() {
+        used_options.push("flake8-type-checking");
+    }
+
+    if top_level_options.flake8_gettext.is_some() {
+        used_options.push("flake8-gettext");
+    }
+
+    if top_level_options.flake8_implicit_str_concat.is_some() {
+        used_options.push("flake8-implicit-str-concat");
+    }
+
+    if top_level_options.flake8_import_conventions.is_some() {
+        used_options.push("flake8-import-conventions");
+    }
+
+    if top_level_options.flake8_pytest_style.is_some() {
+        used_options.push("flake8-pytest-style");
+    }
+
+    if top_level_options.flake8_unused_arguments.is_some() {
+        used_options.push("flake8-unused-arguments");
+    }
+
+    if top_level_options.isort.is_some() {
+        used_options.push("isort");
+    }
+
+    if top_level_options.mccabe.is_some() {
+        used_options.push("mccabe");
+    }
+
+    if top_level_options.pep8_naming.is_some() {
+        used_options.push("pep8-naming");
+    }
+
+    if top_level_options.pycodestyle.is_some() {
+        used_options.push("pycodestyle");
+    }
+
+    if top_level_options.pydocstyle.is_some() {
+        used_options.push("pydocstyle");
+    }
+
+    if top_level_options.pyflakes.is_some() {
+        used_options.push("pyflakes");
+    }
+
+    if top_level_options.pylint.is_some() {
+        used_options.push("pylint");
+    }
+
+    if top_level_options.pyupgrade.is_some() {
+        used_options.push("pyupgrade");
+    }
+
+    if top_level_options.per_file_ignores.is_some() {
+        used_options.push("per-file-ignores");
+    }
+
+    if top_level_options.extend_per_file_ignores.is_some() {
+        used_options.push("extend-per-file-ignores");
+    }
+
+    if used_options.is_empty() {
+        return;
+    }
+
+    let options_mapping = used_options
+        .iter()
+        .map(|option| format!("{option} -> lint.{option}"))
+        .join(",\n  ");
+
+    warn_user!(
```

Comment:
It's a bit unfortunate that we don't know the configuration file name (relative to the project directory)

Reply:
We might be able to get it... Want me to try?

Reply:
Feel free to do it. It can be done as a separate PR because I'm fairly certain we already emit other warnings without mentioning the configuration file name.